_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,78 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The aim of this project is to automatically create whitelists (to be utilized
+by [vulture](https://github.com/jendrikseipp/vulture)) for projects using sip
+files to communicate between C++ level API and Python programs, most
+significant of them being PyQt projects.
+
+#### IDEAS
+
+* Create a script which takes sip files as input and spits a whitelist with all
```

Comment:
It would be useful to also develop a SIP Bear (and .sbf?) as a background task during your project - they can just be simple pass/fail based on `sipconfig` parser failing, but ideally becomes a little more functional over the course of your project.

Reply:
Let me quote your email here because I guess, the mailing list isn't public:

> I doubt that it'd be useful - SIP is rarely used outside of PyQt.
>
> As for PyQt, it's pretty much a one-man show which doesn't really
accommodate external contributors unfortunately - for example, no public
repository exists.
>
> Also, they're mostly auto-generated (in a half-automatic way) as far as
I'm aware - the .sip format is pretty much a C++ header file (.h) with
additional annotations, so projects have some way of keeping those two
in sync.
>
> I think in case of PyQt, the author has some kind of IDE which parses .h
files, asks him for the annotations for any new functions (when a new Qt
version gets released) and then spits out the .sip files.